### DISCUSSION TITLE
Where mermaid syntax is defined?

### DISCUSSION THREAD

**Discussion by Hi-Angel** [mermaid-js/mermaid#6077] [General]
I noticed Emacs `mermaid-mode` *(Mermaid syntax highlight and indentation)* highlights keywords like `flowchart` and `graph` anywhere in text. But from looking at various Mermaid examples, my feeling is that these are only keywords when they appear at "top-level" *(i.e. when they're are the first word on the line, with optional whitespace before them)*.

But before changing this I need some documentation on mermaid syntax, or perhaps reference to its source code where it does parsing *(because obviously I don't want my changes break something)*. But so far I haven't been able to find anything at all. There are pages [like this one](https://mermaid.js.org/syntax/flowchart.html), but they more of a "tutorial" or "intro", they don't define what exactly syntax is allowed.

**Comment by kairi003** [comment]
I don't know much about it, but these docs might be helpful: [Adding a New Diagram/Chart (Deprecated)](https://mermaid.js.org/community/new-diagram-jison.html), [Adding a New Diagram/Chart](https://mermaid.js.org/community/new-diagram.html)

Older parsers are defined in jison format (e.g. [flow.jison](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/diagrams/flowchart/parser/flow.jison)), and newer ones seem to use langium format (e.g. [packet.langium](https://github.com/mermaid-js/mermaid/blob/develop/packages/parser/src/language/packet/packet.langium))
